### ISSUE TITLE
bigtable conditionFilter with chainfilters as predicate

### ISSUE THREAD

**Issue by justinwp** [GoogleCloudPlatform/google-cloud-go#610]
I'm not sure if I'm misunderstanding something with ChainFilters, but the following test case I created does not pass:

```
{
desc:   "read with chainfilters as condition, strip values on true",
rr:     RowRange{},
filter: ConditionFilter(ChainFilters(ColumnFilter(".*j.*"), ColumnFilter(".*mckinley.*")), StripValueFilter(), nil),
want:   "tjefferson-wmckinley-,wmckinley-tjefferson-",
},
```

```
bigtable_test.go:345: read with chainfilters as condition, strip values on true: wrong reads.
got "gwashington-jadams-,jadams-gwashington-,jadams-tjefferson-,tjefferson-gwashington-,tjefferson-jadams-,tjefferson-wmckinley-,wmckinley-tjefferson-"
want "tjefferson-wmckinley-,wmckinley-tjefferson-"
```

**Comment by garye** [comment]
First of all, this test case is wrong when run against the emulator/fake server leading to the output you pasted here. There's a bug in there somewhere. When I run this against a live instance (`go test -it.use-prod=true -it.project=<proj> -it.instance=<instance> -it.table=test-table` it gets no results.

This test case is looking for rows with columns that have a "j" AND "mckinley" in the name, of which there are none. Perhaps there's just some confusion about the test data itself. Can you describe what you want the filter to do?

**Comment by justinwp** [comment]
Sorry for not looking at the test data close enough. My use case chain filter is match row and match the value in those rows.

**Comment by garye** [comment]
If you need to find rows with a certain row key AND a certain column value, you can chain together a `RowKeyFilter` with a `ColumnFilter`, or use the `RowSet` parameter to `ReadRows` instead of the `RowKeyFilter` and add a `ColumnFilter`. This works if you don't need a regexp to find your row keys and Bigtable won't have to do a full table scan.

There's a lot more info about filters here: https://github.com/googleapis/googleapis/blob/master/google/bigtable/v2/data.proto#L179. If something is useful there but isn't present in the Go library docs let me know. Also, if the behavior seems strange against the emulator but works against a live instance (using the incantation above) feel free to file another bug here. I'll keep this open while we fix the original issue you reported.

**Comment by garye** [comment]
FYI, fix for that original issue: https://code-review.googlesource.com/12537

**Comment by garye** [comment]
Fixed by c10801b.
